src/binder/textualidentityparser.cpp

```diff
@@ -417,12 +439,64 @@ namespace BINDER_SPACE
                 cursor++;
             }
 
-            if (dwFoundNumbers == static_cast<DWORD>(iRequiredVersionParts))
+            // Compat with .NET Framework:
+            // - If no value is provided, the version is considered invalid.
+            //   Example: "MyAssembly, Version="
+            // - If the major version is provided and the remaining components are valid, the version is considered valid.
+            //   Examples:
+            //     "MyAssembly, Version=1"
+            //     "MyAssembly, Version=1."
+            //     "MyAssembly, Version=1.."
+            //     "MyAssembly, Version=1..."
+            if (dwFoundNumbers == 0)
             {
-                pAssemblyVersion->SetFeatureVersion(dwNumbers[0], dwNumbers[1]);
-                pAssemblyVersion->SetServiceVersion(dwNumbers[2], dwNumbers[3]);
-                fIsValid = TRUE;
+                goto Exit;
             }
+
+            if (dwFoundNumbers < 2 || dwNumbers[0] == UnspecifiedNumber || dwNumbers[1] == UnspecifiedNumber)
+            {
+                // Compat with .NET Framework: If the major or minor version is unspecified, all components are considered
+                // unspecified. Set the major version to the unspecified value (this is a special value indicating that the
+                // whole version is unspecified, see AssemblyName::HaveAssemblyVersion).
+                //
+                // Examples:
+                //   "MyAssembly, Version=."
+                //   "MyAssembly, Version=.1"
+                //   "MyAssembly, Version=1"
+                //   "MyAssembly, Version=1."
+                dwNumbers[0] = UnspecifiedNumber;
+                dwNumbers[1] = 0;
+                dwNumbers[2] = 0;
+                dwNumbers[3] = 0;
+            }
+            else
+            {
+                // Difference from .NET Framework:
+                // - In .NET Framework, unspecified components retain the unspecified value. This special value is treated as
+                //   unspecified when loading an assembly, even if explicitly specified as 65535 as part of the version.
+                // - In .NET Core, any unspecified version component after the minor version is given a value of zero. When
+                //   loading an assembly, .NET Core will match and load an assembly that has a version that is >= the requested
```

Comment:
@AtsushiKan, it seemed to me like 1.1 and 1.1.0 both represent the same desired version 1.1.0.0, I'm thinking users would expect the former two to be shortcuts to the latter.

I believe it also matches the expected loading behavior. If I call:
```c#
Assembly.Load(new AssemblyName("Foo, Version=1.1"));
```
And the actual assembly's version is 1.1.0.0, it should be loaded. If we turn 1.1 into 1.1.65535.65535, we could also change the loader to treat 65535 as a special value to get the same loading behavior, but I don't see any advantages from doing this. It seems like .NET Framework needed the special value for when it wants to match the exact version while loading, but still be able to ignore parts of the version that were not specified.